contracts/TownHallZap.sol

```diff
@@ -28,4 +36,41 @@ contract TownHallZap {
             townHall.mint(to);
         }
     }
+
```

Comment:
Can we also implement `estimateAmountIn(address sourceToken) external view returns (uint256)` function so the front-end can calculate `amountInMaximum` parameter before calling `convertAndMint()`?

Reply:
It seems we can implement `estimateAmountIn` using `IQuoter` 
ref: https://github.com/Uniswap/v3-periphery/blob/main/contracts/interfaces/IQuoter.sol

Reply:
thx.  I'm applying and correcting it.

Reply:
Hum.. maybe we can't use `IQuoter`.
On https://docs.uniswap.org/contracts/v3/reference/periphery/interfaces/IQuoter it says:
> These functions are not marked view because they rely on calling non-view functions and reverting to compute the result. They are also not gas efficient and should not be called on-chain.

Reply:
uhm...I will look for it and commended if I find it.

Reply:
https://soliditydeveloper.com/uniswap3

Is it possible to look at the picture and use it in the frontend?

<img width="1414" alt="스크린샷 2022-12-29 오후 6 37 33" src="https://user-images.githubusercontent.com/77770252/209932695-0475d8ba-dec5-4678-a788-68ee6ca1d575.png">

Reply:
Ok it seems we should use `callStatic` function on Quoter
ref: https://docs.uniswap.org/sdk/v3/guides/creating-a-trade#using-callstatic-to-return-a-quote